### PR TITLE
Fixing bugs filed in bug bash

### PR DESCRIPTION
**Pull request by yifguoMSFT**
Fixed 2 null reference exception and other minor fixes

### CODE REVIEW

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/configFailureFlow.js [hunk 1]
```diff
@@ -86,7 +92,29 @@ export var configFailureFlow = {
     }
 }
 
-async function checkSubnetAvailabilityAsync(siteInfo, diagProvider, subnetData) {
+async function getAspSites(diagProvider, serverFarmId) {
+    var views = [];
+    var aspSitesObj = await diagProvider.getArmResourceAsync(serverFarmId + "/sites");
+    
+    if (aspSitesObj.status == 401) {
```
Comment (cindylovescoding) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/configFailureFlow.js:99]: Do we want to check if aspSitesObj is null?

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/configFailureFlow.js [hunk 2]
```diff
@@ -208,27 +234,29 @@ async function checkSubnetAvailabilityAsync(siteInfo, diagProvider, subnetData)
...
-            successMsg += `<tr><td>${subnetSize}</td><td>${aspSku}</td><td><b>/27</b></td><td>32-5 = <b>27</b> Addresses</td></tr>`;
-            successMsg += `<tr><td colspan='4'><i>Note: Azure reserves 5 IP addresses within each subnet.</i></td></tr></table>`;
-            successMsg += `<u>Steps to increase the subnet size:</u>`;
-            successMsg += `<li>In this App Service Plan, disconnect all the Web Apps that are currently using Regional VNet integration.</li>`;
-            successMsg += `<li>Increase the subnet size as per the recommendations.</li>`;
-            successMsg += `<li>Reconnect the webapps to the same subnet.</li>`;
+        var aspSku = aspData["sku"] && aspData["sku"]["name"];
+
+        if (aspSku != null) {
+            if (subnetSize > 26 & aspSku[0] == "P") {
```
Comment (cindylovescoding) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/configFailureFlow.js:240]: Let's try to use "===" for comparison as much as possible:)

**Review by cindylovescoding — APPROVED** [review]

**Review by ShekharGupta1988 — APPROVED** [review]

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/configFailureFlow.js (hunk 1 above)
Comment (yifguoMSFT) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/configFailureFlow.js:99]: Nope, the return object of `getArmResourceAsync` should be not null

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/configFailureFlow.js (hunk 2 above)
Comment (yifguoMSFT) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/configFailureFlow.js:240]: Replaced most of `==` by `===` other than `==null` or `==number`

**Pull request merged**